assets/src/hooks/useChannel.ts

```diff
@@ -0,0 +1,64 @@
+import { Channel, Socket } from "phoenix"
+import { useEffect, useState } from "react"
+import { reload } from "../models/browser"
+
+/** Opens a channel for the given topic
+ *  and returns the latest data that's been pushed to it
+ *
+ *  Only listens for a single event
+ *  Assumes the data returned on join and the data from subsequent pushes are the same format
+ *  If topic is null, does not open a channel.
+ *  Returns offState if the channel is not open yet
+ *  Returns loadingState after the channel has been opened but before receiving the first response
+ */
+export const useChannel = <T>({
+  socket,
+  topic,
+  event,
+  parser,
+  loadingState,
+  offState,
+}: {
+  socket: Socket | undefined
+  topic: string | null
+  event: string
+  parser: (data: any) => T
+  loadingState: T
+  offState: T
```

Comment:
I'm confused by what `offState` means. Can we just use `loadingState`?

Reply:
`useSearchResults` returns a different value to distinguish between "we're waiting for a response (`loadingState`,`null`) and "we haven't initialized the socket yet"/"we're not searching (`offState`,`undefined`), but then we don't do anything interesting with those values. I'll consolidate them.